src/components/Calculator.js

```diff
@@ -1,35 +1,54 @@
 /* eslint-disable */
 import { Component } from 'react';
-import Cell from './Cell';
+import calculate from '../logic/calculate'
 
 class Calculator extends Component {
   constructor(props) {
     super(props);
+    this.state = {
+      total: 0,
+      next: null,
+      operation: null
+    };
+    this.calculateHandler = this.calculateHandler.bind(this);
+  }
+
+  calculateHandler (event) {
+    
+    this.setState((state) => (calculate(state, event.target.innerHTML)));
   }
 
   render() {
+    const { total, next, operation } = this.state;
     return (
       <div className="grid-container">
-        <Cell className="result-field" text="0" />
-        <Cell text="AC" />
-        <Cell text="+/-" />
-        <Cell text="%" />
-        <Cell className="main-operation" text="÷" />
-        <Cell text="7" />
-        <Cell text="8" />
-        <Cell text="9" />
-        <Cell className="main-operation" text="X" />
-        <Cell text="4" />
-        <Cell text="5" />
-        <Cell text="6" />
-        <Cell className="main-operation" text="-" />
-        <Cell text="1" />
-        <Cell text="2" />
-        <Cell text="3" />
-        <Cell className="main-operation" text="+" />
-        <Cell className="zero" text="0" />
-        <Cell text="." />
-        <Cell className="main-operation" text="=" />
+      
+        <div className="result-field" text="0" >
+        { total }
+        { operation }
+        { next }
+        </div>
+
+        <button className="cell" onClick={this.calculateHandler}>AC</button>
+        <button className="cell" onClick={this.calculateHandler}>+/-</button>
+        <button className="cell" onClick={this.calculateHandler}>%</button>
+        <button className="main-operation" onClick={this.calculateHandler}>÷</button>
+        <button className="cell" onClick={this.calculateHandler}>7</button>
+        <button className="cell" onClick={this.calculateHandler}>8</button>
+        <button className="cell" onClick={this.calculateHandler}>9</button>
+        <button className="main-operation" onClick={this.calculateHandler}>X</button>
```

Comment:
Hi @devMahmoud 
- [ ] Please do your best to fix the multiplication button. The app generates an error when users try to multiply two numbers and click the equal sign for the result.
![error](https://user-images.githubusercontent.com/55596970/170609209-17bfda1a-cfdf-4965-be70-9fdaf1f43fb9.png)